src/fidesops/task/graph_task.py

```diff
@@ -226,9 +233,23 @@ def log_end(
     @retry(action_type=ActionType.access, default_return=[])
     def access_request(self, *inputs: List[Row]) -> List[Row]:
         """Run access request"""
+        formatted_input_data = self.to_dask_input_data(*inputs)
         output = self.connector.retrieve_data(
-            self.traversal_node, self.resources.policy, self.to_dask_input_data(*inputs)
+            self.traversal_node, self.resources.policy, formatted_input_data
         )
+
+        for row in output:
+            # In code, filter out non-matching sub-documents and array elements
+            logger.info(
+                f"Filtering row in {self.traversal_node.node.address} for matching array elements."
+            )
+            filter_element_match(
+                row,
+                {
+                    FieldPath.parse(field): inputs
+                    for field, inputs in formatted_input_data.items()
+                },
+            )
```

Comment:
This is a primary change: I filter out non-matching sub-documents and array elements in code, this is also a function that we may make configurable https://github.com/ethyca/fidesops/issues/193, although I think the default behavior makes sense, in that it views array data as nested documents, and only matched nested documents are used to find subsequent documents.